src/connect.js

```diff
@@ -85,33 +108,10 @@ const verifyReactImpureComponent = (target, prop, descriptor) => {
   if (isReactPureComponent(target)) {
     throw new TypeError('@connect should not be used for pure component.');
   }
-  if (isReactComponent(target)) {
-    return true;
-  }
-  return false;
-};
-
-const connectWithCursor = cursorPaths => (target, prop, descriptor) => {
-  if (!target) {
-    throw new TypeError('connect(...)() is invalid, the param component must be given.');
-  }
-  if (verifyReactImpureComponent(target, prop, descriptor)) {
-    connectWrapper(target, cursorPaths);
-  } else {
-    throw new TypeError('@connect must be used for component.');
-  }
-};
-
-const connect = (target, prop, descriptor) => {
-  if (!target) {
-    throw new TypeError('@connect() is invalid, do you mean @connect or @connect(\'\') or @connect([]) ?');
-  }
-  if (verifyReactImpureComponent(target, prop, descriptor)) {
-    return connectWrapper(target);
-  } else {
-    const cursorPaths = Array.isArray(target) ? target : [target];
-    return connectWithCursor(cursorPaths);
+  if (!isReactComponent(target)) {
```

Comment:
A good suggestion but many details need to be discussed but not this pull request.